tests/e2e/test_spyre_cb_scheduler_steps.py

```diff
@@ -764,10 +764,15 @@ def test_new_sequence_joins_during_decode(model: str, backend: str,
 @pytest.mark.cb
 @pytest.mark.parametrize("model", get_spyre_model_list())
 @pytest.mark.parametrize("backend", get_spyre_backend_list())
+@pytest.mark.parametrize("prefill_optimization", [True, False])
 def test_prompt_too_long_for_current_tkv(model: str, backend: str,
+                                         prefill_optimization: bool,
                                          monkeypatch: pytest.MonkeyPatch,
                                          set_random_seed):
     """ Scenario where the requested prompt is too long for current tkv value
+   
+     Note that with VLLM_SPYRE_ENABLE_PREFILL_OPTIMIZATION enabled, we can 
+    prefill the prompt straight away -> using checked_steps_with_optimization
```

Comment:
It'd be great to have a negative test as well for the case where we can't bump the tkv because the remaining tokens to be generated of a running sequence would overrun the max context length if we did